meta1v2/meta1_gridd_dispatcher.c

```diff
@@ -46,13 +46,29 @@ static GByteArray *encode_and_clean(GByteArray* (*e)(gchar**), gchar **pv) {
 	return result;
 }
 
+static struct oio_url_s * _extract_url(struct gridd_reply_ctx_s *reply)
+{
+	struct oio_url_s *url = metautils_message_extract_url (reply->request);
+
+	const gchar *_err;
+	if (!oio_url_check(url, NULL, &_err)) {
+		reply->send_error(0, BADREQ("Invalid %s", _err));
+		oio_url_clean(url);
+		return NULL;
+	}
+	return url;
+}
+
 /* -------------------------------------------------------------------------- */
 
 static gboolean
 meta1_dispatch_v2_USERCREATE(struct gridd_reply_ctx_s *reply,
 		struct meta1_backend_s *m1, gpointer ignored UNUSED)
 {
-	struct oio_url_s *url = metautils_message_extract_url (reply->request);
+	struct oio_url_s *url = _extract_url(reply);
+	if (!url) {
+		return TRUE;
```

Comment:
It is highly recommended either to reply something to the client or to abruptly close the connection with a `return FALSE`.